crates/ethereum/evm/src/execute.rs

```diff
@@ -178,7 +179,13 @@ where
                 .into())
             }
 
-            self.evm_config.fill_tx_env(evm.tx_mut(), transaction, *sender);
+            self.evm_config.fill_tx_env(
+                evm.tx_mut(),
+                &TransactionSignedEcRecovered::from_signed_transaction(
+                    transaction.clone(),
+                    *sender,
```

Comment:
I feel like this is a bit too expensive for now,

I want to solve this problem differently and perhaps only restrict the tx type to a trait on ethexecutor level